tests/cypress/integration/reportRepair/repairProblem.spec.js

```diff
@@ -1,250 +1,65 @@
 import {
+  checkIfSelectionGoesToCorrectUrl,
+  makeSelectionAndClickButton,
   navigateToLocation
 } from '../../support/helpers';
 
-const genericWhatIsTheProblemOptions = () => {
-  it('displays the repair problem question', () => {
-    cy.contains('What is the problem?');
-  });
-
-  it('displays a "walls" option', () => {
-    cy.contains('Walls, floor or ceiling, excluding damp');
-  });
-
-  it('displays a "electrics" option', () => {
-    cy.contains('Electrics, including lights and switches');
-  });
-
-}
 describe('repairProblem', () => {
 
-  context('kitchen', () => {
+  context('When all options are displayed and a user doesn\'t select anything', () => {
     before(() => {
       navigateToLocation()
-      cy.contains('Kitchen').click();
-      cy.get('button').click();
+      makeSelectionAndClickButton('Kitchen');
     });
 
     it('displays the repair problem question', () => {
       cy.contains('What is the problem?');
     });
 
-    it('displays a "cupboards" option', () => {
-      cy.contains('Cupboards, including damaged cupboard doors');
-    });
-
-    it('displays a "walls" option', () => {
-      cy.contains('Walls, floor or ceiling, excluding damp');
-    });
-
-    it('displays a "electrical" option', () => {
-      cy.contains('Electrical, including extractor fans and lightbulbs');
-    });
-
-    it('displays a "worktop" option', () => {
-      cy.contains('Damaged worktop');
-    });
-
-    it('displays a "sink" option', () => {
-      cy.contains('Sink, including taps and drainage');
-    });
-
-    it('displays a "door" option', () => {
-      cy.contains('Damaged or stuck doors');
-    });
-
-    it('displays a "heating" option', () => {
-      cy.contains('Heating or hot water');
-    });
-
-    it('displays a "window" option', () => {
-      cy.contains('Damaged or stuck windows');
-    });
-
-    it('displays a "Damp or mould" option', () => {
-      cy.contains('Damp or mould');
-    });
-
-    context('When a user doesn\'t select anything', ()=>{
-      it('should show validation message',  () => {
-        cy.get('button').click().then(()=>{
-          cy.contains('Select the problem you are reporting');
-        });
+    it('displays all the options', () => {
+      const options = [
+        'Cupboards, including damaged cupboard doors',
+        'Electrical, including extractor fans and lightbulbs',
+        'Damaged worktop',
+        'Heating or hot water',
+        'Damaged or stuck doors',
+        'Walls, floor or ceiling, excluding damp',
+        'Sink, including taps and drainage',
+        'Damaged or stuck windows',
+        'Damp or mould'
+      ]
+      options.forEach((option) => {cy.contains(option)})
```

Comment:
This will fail for the first item not found and not attempt the others which _could_ mean repeated iterations of tests to find additional test failures.

Try to find a way to run all these cases in a single run.
Alternatively, restore to the original tests

Reply:
It makes sense. I've changed the loop to generate the whole tests insise:

```
options.forEach((option) => {
      it(`Displays '${option}' option`, () => {
        cy.contains(option)
      })
    })
```